perf_guide.md

```diff
@@ -98,13 +130,23 @@ implementations at run-time without issues (e.g. using `#[inline(always)]` for
 the impls, wrapping in `#[target_feature]`, and the wrapping those in a function
 that does run-time feature detection).
 
+**NOTE** (x86 specific): because the AVX (256-bit) registers extend the existing
+SSE (128-bit) registers, mixing SSE and AVX instructions in a program can cause
+performance issues.
+
+The solution is to compile all code, even the code written with 128-bit vectors,
+with the AVX target feature enabled. This will cause the compiler to prefix the
+generated instructions with the [VEX] prefix.
+
+[VEX]: https://en.wikipedia.org/wiki/VEX_prefix
+
```

Comment:
So I think we should add a section here with a TODO about how to do run-time feature detection. And one afterwards about how to combine everything that we have learned.

The structure of this would then be something like: Use `RUSTFLAGS` to enable feature globally, learn how to use `#[target_feature]` to embed code for different targets into the same binary, learn how to use `is_{ARCH}_feature_detected` to detect which features the CPU supports, learn how to combine `#[target_feature]` with `is_{ARCH}_feature_detected` to write code that runs everywhere (e.g. there is an idiom of writing the code using intrinsics in an `#[inline(always)]` function, and then building complex algorithms in `#[inline(always)]` functions, and at the top level, wrap the whole algorithm into different `#[target_feature]` functions that get dispatched using `is_..._feature_detected`.